balancer-js/examples/pools/composable-stable/init-join.ts

```diff
@@ -0,0 +1,70 @@
+import { Interface, LogDescription } from '@ethersproject/abi';
+import { parseFixed } from '@ethersproject/bignumber';
+import { Contract } from '@ethersproject/contracts';
+import { Log, TransactionReceipt } from '@ethersproject/providers';
+
+import create, {
+  addresses,
+  balancer,
+  provider,
+  signer,
+  tokenAddresses,
+} from './create';
+import { Vault__factory } from '@/contracts/factories/Vault__factory';
+import ComposableStablePoolAbi from '@/lib/abi/ComposableStable.json';
+import { isSameAddress } from '@/lib/utils';
+import { PoolType } from '@/types';
+
+const initJoinComposableStable = async () => {
+  const poolAddress = await create;
+  const signerAddress = await signer.getAddress();
+  const composableStablePoolInterface = new Interface(ComposableStablePoolAbi);
+  const pool = new Contract(
+    poolAddress,
+    composableStablePoolInterface,
+    provider
+  );
+  const composableStablePoolFactory = balancer.pools.poolFactory.of(
+    PoolType.ComposableStable
+  );
+  const amountsIn = [
+    parseFixed('10000', addresses.USDC.decimals).toString(),
+    parseFixed('10000', addresses.USDT.decimals).toString(),
+  ];
+  const poolId = await pool.getPoolId();
+  const { to, data } = composableStablePoolFactory.buildInitJoin({
+    joiner: signerAddress,
+    poolId,
+    poolAddress,
+    tokensIn: tokenAddresses,
+    amountsIn,
+  });
+  const tx = await signer.sendTransaction({
+    to,
+    data,
+    gasLimit: 30000000,
+  });
+  await tx.wait();
+  const receipt: TransactionReceipt = await provider.getTransactionReceipt(
+    tx.hash
+  );
+  const vaultInterface = new Interface(Vault__factory.abi);
+  const poolInitJoinEvent: LogDescription | null | undefined = receipt.logs
```

Comment:
Probably easier/clearer to just check balances?

Reply:
I couldn't verify the balances using the getBalances, because in the time I check the balances of the pool, they still 0, the balanceDeltas of the vault for the tokens I sent are exactly the same value I sent though(even after a timeout of 30 seconds), the Vault holds the tokens of the init join for some reason